patina_dxe_core/src/gcd/spin_locked_gcd.rs

```diff
@@ -421,6 +421,7 @@ impl GCD {
         ensure!(self.maximum_address != 0, EfiError::NotReady);
         ensure!(len > 0, EfiError::InvalidParameter);
         ensure!(base_address.checked_add(len).is_some_and(|sum| sum <= self.maximum_address), EfiError::Unsupported);
+        ensure!(self.memory_blocks.capacity() > 0, EfiError::NotReady);
```

Comment:
Is a unit test already checking the `self.memory_blocks.capacity()==0` condition?